app/controllers/admin/collections_controller.rb

```diff
@@ -217,4 +217,40 @@ def destroy
       redirect_to admin_collection_path(@source_collection)
     end
   end
+
+  # POST /collections/1/batch_ingest
+  def batch_ingest
+    if @collection.dropbox.manifest_processable?
+      # Spawn off background job
+      AvalonJobs.collection_batch_ingest(@collection.id)
+      flash[:notice] = t('batch_ingest.initiated', name: @collection.name)
+    else
+      flash[:error] =
+        if @collection.dropbox.manifest_error?
```

Comment:
I agree with making the error message better and more contained in the dropbox class to avoid the branching logic. :+1:
 
I don't really agree with using a status value over using the predicate functions. The predicate functions are also used in the test suite to test that certain conditions are set up correctly, and I would prefer to expose "manifest_error?" over exposing "manifest_status == :error" outside of the dropbox class.